app/models/request.rb

```diff
@@ -38,7 +38,22 @@ def stats
     }
   end
 
+  def messages_by_user
+    messages
+      .where(broadcasted: false)
+      .group_by(&:user)
+      .transform_values { |messages| messages.sort_by(&:created_at) }
```

Comment:
That's not necessary if you would `reorder(created_at: :asc)`.

Reply:
https://stackoverflow.com/questions/11350500/ruby-hash-keys-and-values-safe-to-assume-same-order

Reply:
The thing is I need them to be sorted desc first, then group, then sort asc inside of the groups. Suggestions?

Reply:
OK, fair enough.